csa-runner-app/src/main/java/com/farao_community/farao/swe_csa/app/dichotomy/SweCsaDichotomyRunner.java

```diff
@@ -0,0 +1,185 @@
+        RaoResponse raoResponseAfterDichotomy = getDichotomyResponse(network, crac, validator, indexStrategy);
+        LOGGER.info("dichotomy RAO computation answer received for TimeStamp: '{}'", raoRequest.getInstant());
+
+        return new CsaResponse(raoResponseAfterDichotomy.getId(), Status.FINISHED.toString());
+    }
+
+    protected RaoResponse getDichotomyResponse(Network network, Crac crac, SweCsaRaoValidator validator, SweCsaHalfRangeDivisionIndexStrategy indexStrategy) {
+        DichotomyResult<RaoResponse, MultipleDichotomyVariables> result = this.getEngine(network, crac, validator, indexStrategy).run(network);
+        return result.getHighestValidStep().getValidationData();
+    }
+
+    protected SweCsaDichotomyEngine getEngine(Network network, Crac crac, SweCsaRaoValidator validator, SweCsaHalfRangeDivisionIndexStrategy indexStrategy) {
+        this.updateCracWithCounterTrageRangeActions(crac);
+        Pair<MultipleDichotomyVariables, MultipleDichotomyVariables> initialDichotomyVariable = getInitialDichotomyIndex(crac);
+
+        return new SweCsaDichotomyEngine(
+            new Index<>(initialDichotomyVariable.getLeft(), initialDichotomyVariable.getRight(), 10),
+            indexStrategy,
+            new LinearScaler(SweCsaZonalData.getZonalData(network), new SweCsaShiftDispatcher(getInitialPositions(crac))),
+            validator);
+    }
+
+    private Map<String, Double>  getInitialPositions(Crac crac) {
+        CounterTradeRangeAction ctRaFrEs = crac.getCounterTradeRangeAction(CounterTradeRangeActionDirection.FR_ES.getName());
+        CounterTradeRangeAction ctRaPtEs = crac.getCounterTradeRangeAction(CounterTradeRangeActionDirection.PT_ES.getName());
+
+        double expFrEs = ctRaFrEs.getInitialSetpoint();
+        double expPtEs = ctRaPtEs.getInitialSetpoint();
+
+        return Map.of(
+            Country.ES.getName(), expFrEs + expPtEs,
+            Country.FR.getName(), -expFrEs,
+            Country.PT.getName(), -expPtEs);
+    }
+
+    private Pair<MultipleDichotomyVariables, MultipleDichotomyVariables> getInitialDichotomyIndex(Crac crac) {
+        CounterTradeRangeAction ctRaFrEs = crac.getCounterTradeRangeAction(CounterTradeRangeActionDirection.FR_ES.getName());
+        CounterTradeRangeAction ctRaEsFr = crac.getCounterTradeRangeAction(CounterTradeRangeActionDirection.ES_FR.getName());
+        CounterTradeRangeAction ctRaPtEs = crac.getCounterTradeRangeAction(CounterTradeRangeActionDirection.PT_ES.getName());
+        CounterTradeRangeAction ctRaEsPt = crac.getCounterTradeRangeAction(CounterTradeRangeActionDirection.ES_PT.getName());
+
+        double expFrEs = ctRaFrEs.getInitialSetpoint();
+        double expPtEs = ctRaPtEs.getInitialSetpoint();
+        double expEsFr = ctRaEsFr.getInitialSetpoint();
+        double expEsPt = ctRaEsPt.getInitialSetpoint();
```

Comment:
initial exchange should be computed using load flow + summing on tielines and hvdc lines. see how they do it for CC